### ISSUE TITLE
problems building openCV as numpy not found

### ISSUE THREAD

**Issue by RichardAW** [numpy/numpy#16521]
Hello, 

I am trying to build openCV for visual studio 2019 using CMake 3.17.2.  I've installed x64 anaconda3 and anaconda 2 and I have updated the packages in both. 

In CMake I receive this error when configuring (source openCV) I've tried reinstalling numpy and tired a different install package. I've also tired uninstalling python 2 but that didn't help. I'm pretty new to this so any help would be much appreciated if you new more information just let me know. Thanks!

### Error message:

Found PythonInterp: C:/DEV/Anaconda2/python.exe (found suitable version "2.7.16", minimum required is "2.7") 
Found PythonLibs: C:/DEV/Anaconda2/libs/python27.lib (found suitable exact version "2.7.16") 
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\DEV\Anaconda2\lib\site-packages\numpy\__init__.py", line 142, in <module>
    from . import core
  File "C:\DEV\Anaconda2\lib\site-packages\numpy\core\__init__.py", line 71, in <module>
    raise ImportError(msg)
ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the multiarray numpy extension module failed.  Most
likely you are trying to import a failed build of numpy.
Here is how to proceed:
- If you're working with a numpy git repository, try `git clean -xdf`
  (removes all files not under version control) and rebuild numpy.
- If you are simply trying to use the numpy version that you have installed:
  your installation is broken - please reinstall numpy.
- If you have already reinstalled and that did not fix the problem, then:
  1. Check that you are using the Python you expect (you're using C:\DEV\Anaconda2\python.exe),
     and that you have no directories in 
your PATH or PYTHONPATH that can
     interfere with the Python and numpy versions you're trying to use.
  2. If (1) looks fine, you can open a new issue at
     https://github.com/numpy/numpy/issues.  Please include details on:
     - how you installed Python
     - how you installed numpy
     - your operating system
     - whether or not you have multiple versions of Python installed
     - if you built from source, your compiler versions and ideally a build log

     Note: this error has many possible causes, so please don't comment on
     an existing issue about this - open a new one instead.

Original error was: DLL load failed: The specified module could not be found.

Found PythonInterp: C:/DEV/Anaconda3/python.exe (found suitable version "3.7.6", minimum required is "3.2") 
Found PythonLibs: C:/DEV/Anaconda3/libs/python37.lib (found suitable exact version "3.7.6") 
Traceback (most recent call last):
  File "C:\DEV\Anaconda3\lib\site-packages\numpy\core\__init__.py", line 17, in <module>
    from . import multiarray
  File "C:\DEV\Anaconda3\lib\site-packages\numpy\core\multiarray.py", line 14, in <module>
    from . import overrides
  File "C:\DEV\Anaconda3\lib\site-packages\numpy\core\overrides.py", line 7, in <module>
    from numpy.core._multiarray_umath import (
ImportError: DLL load failed: The specified module could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\DEV\Anaconda3\lib\site-packages\numpy\__init__.py", line 142, in <module>
    from . import core
  File "C:\DEV\Anaconda3\lib\site-packages\numpy\core\__init__.py", line 47, in <module>
    raise ImportError(msg)
ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy c-extensions failed.
- Try uninstalling and reinstal
ling numpy.
- If you have already done that, then:
  1. Check that you expected to use Python3.7 from "C:\DEV\Anaconda3\python.exe",
     and that you have no directories in your PATH or PYTHONPATH that can
     interfere with the Python and numpy version "1.17.5" you're trying to use.
  2. If (1) looks fine, you can open a new issue at
     https://github.com/numpy/numpy/issues.  Please include details on:
     - how you installed Python
     - how you installed numpy
     - your operating system
     - whether or not you have multiple versions of Python installed
     - if you built from source, your compiler versions and ideally a build log

- If you're working with a numpy git repository, try `git clean -xdf`
  (removes all files not under version control) and rebuild numpy.

Note: this error has many possible causes, so please don't comment on
an existing issue about this - open a new one instead.

Original error was: DLL load failed: The specified module could not be found.

### Numpy/Python version information:
numpy ver 1.18.1
python 3.7.6

**Comment by mattip** [comment]
Is this using `C:/DEV/Anaconda2/python.exe` or `C:/DEV/Anaconda3/python.exe` ? There seems to be two error messages mixed together. In any case, you may need to [activate your conda environment](https://numpy.org/devdocs/user/troubleshooting-importerror.html) before calling CMake.

**Comment by RichardAW** [comment]
Hello, thanks for your reply. I think it's using both? I get an error when I only have python3. Should I delete python2? 

I have added both C:\DEV\anaconda3\Library\bin and C:\DEV\anaconda2\Library\bin to my path in environment variables which seems to have removed this issue. 

**CMake output at configure:** 

_Detecting C compiler ABI info
Detecting C compiler ABI info - done
Detecting C compile features
Detecting C compile features - done
Detected processor: AMD64
Found PythonInterp: C:/DEV/Anaconda2/python.exe (found suitable version "2.7.16", minimum required is "2.7") 
Found PythonLibs: C:/DEV/Anaconda2/libs/python27.lib (found suitable exact version "2.7.16") 
Found PythonInterp: C:/DEV/anaconda3/python.exe (found suitable version "3.7.6", minimum required is "3.2") 
Found PythonLibs: C:/DEV/anaconda3/libs/python37.lib (found suitable exact version "3.7.6") 
Performing Test HAVE_CXX_FP:PRECISE
Performing Test HAVE_CXX_FP:PRECISE - Success
Performing Test HAVE_C_FP:PRECISE
Performing Test HAVE_C_FP:PRECISE - Success_


**though I now have a new error:**

_CMake Warning at cmake/OpenCVGenSetupVars.cmake:54 (message):
  CONFIGURATION IS NOT SUPPORTED: validate setupvars script in install
  directory
Call Stack (most recent call first):
  CMakeLists.txt:967 (include)_


I tried building in visual studio and I get thousands of errors. 


Thanks, 
Richard
